File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        while (s.length() > 0) {
            char x = s.charAt(0);
            int x_same = 0;
            int x_other = 0;
            int count = 0;
            for (char c : s.toCharArray()) {
                if (c == x) {
                    x_same++;
                    count++;
                } else {
                    x_other++;
                    count++;
                }
                if (x_same == x_other)
                    break;
            }
            s = s.substring(count);

            answer++;
        }

        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    x = ''
    x_count = 0
    other_count = 0

    for c in s:
        if x == '':
            x = c

        if c == x:
            x_count += 1
        else:
            other_count += 1

        if x_count == other_count:
            answer += 1
            x = ''
            x_count = 0
            other_count = 0

    if x_count != 0 or other_count != 0:
        answer += 1

    return answer